codebuild/cd/generic-unix-build.sh

```diff
@@ -43,3 +36,10 @@ JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64 mvn -B package -DskipTests -Dshared
 
 aws s3 cp --recursive --include "*.so" target/cmake-build/lib s3://aws-crt-java-pipeline/${GIT_TAG}/lib
 aws s3 cp target/ s3://aws-crt-java-pipeline/${GIT_TAG}/jar/ --recursive --exclude "*" --include "aws-crt*.jar"
+
+if [[ $AWS_CRT_TARGET != linux-armv8 ]]; then
+    # This script used to compile for linux-armv6, linux-armv7 and linux-armv8 with no-fips
+    # Only when compile for linux-armv8, the shared lib should only be uploaded to /lib and not for /fips_lib
+    # Because: the FIPS subset contains FIPS build for linux-armv8 and linux-x64, and NON-FIPS build for all other platforms.
+    aws s3 cp --recursive --include "*.so" target/cmake-build/lib s3://aws-crt-java-pipeline/${GIT_TAG}/fips_lib
```

Comment:
isn't linux-x86_64 also FIPS-compliant and use `generic-unix-build.sh`? I'm weirded out by `generic-unix-build.sh` being like "obviously I'm only used for arm builds"

related: this release process is so complicated, I'm worried we'll accidentally end up with a jar where the things we expected to be FIPS-compliant were accidentally overwritten by non-compliant libs.

Can we do a safety check to the final jar before it goes to the production Maven server? Ensure that all platforms we expected to be FIPS-compliant actually are?

And/Or can we simplify the release process so there's 1 clear place that sorts the fips-compliant from the non-fips-compliant. Like, instead of a bunch of different scripts copying libs into both `s3://.../lib/.` and `s3://.../fips_lib/.`. What if there was just 1 script that handled sorting? So the N scripts that upload their lib, do it to 1 place: lib/ or fips_lib/, as appropriate. Then another script does like:
* download everything from `s3://.../lib/`
* download everything from `s3://.../fips_lib/`
* locally, copy files from `lib/` to `fips_compat_lib/`
* locally, copy files from `fips_lib/` to `fips_compat_lib/`
* use `lib/` for regular uber-jar
* use `fips_compat_lib/` for `fips-compat` uber-jar classifier
* **BONUS POINTS**: use `fips_lib/` for `fips` uber-jar classifer that doesn't contain any non-compliant platforms